### PR TITLE
acquisition: create acquisition resources dumpers

### PR DESCRIPTION
**Pull request by zannkukai**
* Creates dumper classes for acquisitions resources. These dumpers will
  be used for order notification generation.
* Creates order notification preview endpoint.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/acq_order_lines/dumpers.py [hunk 1]
```diff
@@ -66,3 +67,29 @@ def dump(self, record, data):
         }
         data['document'] = {k: v for k, v in data['document'].items() if v}
         return data
+
+
+class AcqOrderLineNotificationDumper(InvenioRecordsDumper):
+    """Order line dumper class for acquisition order."""
+
+    def dump(self, record, data):
+        """Dump an AcqOrderLine instance for acquisition order notification.
```
Comment (iGormilhit) [inline rero_ils/modules/acq_order_lines/dumpers.py:76]:
```suggestion
        """Dump an acquisition order line instance for acquisition order notification.
```

I do prefer natural language in documentation sentences. :slightly_smiling_face: Of course, my opinion stays the same for other occurrences.

**Review by iGormilhit — COMMENTED** [review]
Commit message approved.

File: rero_ils/modules/acq_order_lines/dumpers.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/acq_order_lines/dumpers.py:76]: It' debate time ! 
My feeling is that I would mention this dumper class should be use with an object 'AcqOrderLine' class instance. If I choose your proposal, when I will read this comment in a couple of year, I could think I could use this dumper with any class representing something like a "line for a acquisition order". Is this a comment, Is this an address block, ...

File: rero_ils/modules/acq_order_lines/dumpers.py [hunk 2]
```diff
@@ -66,3 +67,29 @@ def dump(self, record, data):
...
+
+        :param record: The record to dump.
+        :param data: The initial dump data passed in by ``record.dumps()``.
+        """
```
Comment (rerowep) [inline rero_ils/modules/acq_order_lines/dumpers.py:80]: If the data is the `record.dumps()`, why we need the data we can do it here:
`data = records.dumps()` ?

File: rero_ils/modules/acq_order_lines/dumpers.py [hunk 3]
```diff
@@ -66,3 +67,29 @@ def dump(self, record, data):
...
+        })
+        data = {k: v for k, v in data.items() if v}
+
+        # Add document informations: formatted title and ISBN identifiers.
+        document = record.document
+        data['document'] = {
+            'title': title_format_text_head(document.get('title', [])),
+            'identifiers': document.get_identifier_values(filters=['bf:Isbn'])
+        }
+        data['document'] = {k: v for k, v in data['document'].items() if v}
```
Comment (rerowep) [inline rero_ils/modules/acq_order_lines/dumpers.py:94]: Do we use this a lot maybe a utils function could be helpful:
` {k: v for k, v in data['document'].items() if v}`

File: rero_ils/modules/acq_orders/dumpers.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            'note': record.get_note(AcqOrderNoteType.VENDOR),
+        })
+        library = Library.get_record_by_pid(record.library_pid)
+        data['library'] = library.dumps(
+            dumper=LibraryAcquisitionNotificationDumper())
+        data['order_lines'] = [
+            order_line.dumps(dumper=AcqOrderLineNotificationDumper())
+            for order_line in record.get_order_lines()
+        ]
+        data = {k: v for k, v in data.items() if v}
```
Comment (rerowep) [inline rero_ils/modules/acq_orders/dumpers.py:52]: See above.

File: rero_ils/modules/acq_orders/dumpers.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class AcqOrderNotificationDumper(InvenioRecordsDumper):
+    """Order dumper class for acquisition order notification."""
+
+    def dump(self, record, data):
+        """Dump an AcqOrder instance for acquisition order notification.
+
+        :param record: The record to dump.
+        :param data: The initial dump data passed in by ``record.dumps()``.
```
Comment (rerowep) [inline rero_ils/modules/acq_orders/dumpers.py:38]: See above.

File: rero_ils/modules/libraries/dumpers.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class LibraryAcquisitionNotificationDumper(InvenioRecordsDumper):
+    """Library dumper class for acquisition order notification."""
+
+    def dump(self, record, data):
+        """Dump a library instance for acquisition order notification.
+
+        :param record: The record to dump.
+        :param data: The initial dump data passed in by ``record.dumps()``.
```
Comment (rerowep) [inline rero_ils/modules/libraries/dumpers.py:33]: See above.

File: rero_ils/modules/libraries/dumpers.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        """
+        data.update({
+            'name': record.get('name'),
+            'address': record.get_address(LibraryAddressType.MAIN_ADDRESS),
+            'shipping_address': record.get_address(
+                LibraryAddressType.SHIPPING_ADDRESS),
+            'billing_address': record.get_address(
+                LibraryAddressType.BILLING_ADDRESS)
+        })
+        data = {k: v for k, v in data.items() if v}
```
Comment (rerowep) [inline rero_ils/modules/libraries/dumpers.py:43]: See above

**Review by lauren-d — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]

**Review by BadrAly — APPROVED** [review]

**Pull request merged**